manifest/manifestlist/manifestlist.go

```diff
@@ -176,7 +183,14 @@ func (m *DeserializedManifestList) MarshalJSON() ([]byte, error) {
 // Payload returns the raw content of the manifest list. The contents can be
 // used to calculate the content identifier.
 func (m DeserializedManifestList) Payload() (string, []byte, error) {
-	return m.MediaType, m.canonical, nil
+	var mediaType string
+	if m.MediaType == "" {
+		mediaType = v1.MediaTypeImageIndex
+	} else {
+		mediaType = m.MediaType
+	}
+
+	return mediaType, m.canonical, nil
```

Comment:
It's been quite a while since I originally drafted this PR; I don't really recall making this part of the change 🙈. I _think_ it's ok (i.e. default to the expected MediaType if none was set), but pointing it out just in case